README.md

```diff
@@ -25,23 +28,90 @@ you have those available.
 
 By default the backend is organized into 3 subprojects:
  - api: handles all routes, authentication, and services
  - datamodel: case classes for data the api operates on
  - database: handles database interaction with models
 
-### Migrations
+#### Migrations
 This project uses [flyway](https://flywaydb.org/) for migrations. The migrations are stored in the database subproject (`database/src/main/resources/db/migrations`). For information on the naming and formatting conventions consult the [documentation](https://flywaydb.org/documentation/migrations#naming). Each migration should be `sql` and generally follows the format of `V<number>__<description>.aql`.
 
 Running migrations and other tasks are managed through `./scripts/migrate`.
 
-### Development workflow
+#### Development workflow
 Usually at least two terminals should be open for doing development. The first terminal is where `sbt` should be run (`./scripts/console sbt`). In this terminal tests can be run, projects compiled, and the server assembled.
 
 The other terminal is where the server should be run `./scripts/server` or other one-off commands. To see changes you made to the API live you will need to first `assemble` the `jar` for the `api` server.
+
+### UI Development
+
+You can find files for the user interface under [`granary-ui/`](./granary-ui). To get set up with Elm, you'll
+want to make sure a few things are on your path. If you followed the [Quick Setup](#quick-setup) instructions, you have these already.
+
+- [`elm-format`](https://github.com/avh4/elm-format)
+- [`elm`](https://guide.elm-lang.org/install/elm.html)
+
+Currently, the entire UI application lives in a single file. The reason for that is the note under
+[Culture Shock](https://guide.elm-lang.org/webapps/structure.html) in the Elm guide, specifically:
+
+> In JavaScript, the longer your file is, the more likely you have some sneaky mutation that will cause a really difficult bug. But in Elm, that is not possible! Your file can be 2000 lines long and that still cannot happen.
+
+That may change as we separate pages.
+
+#### First steps in this app
+
+Like any of our recent frontend applications, start with `nvm use`. Whenever you make changes that you'd
+like to see served, run `./scripts/update --frontend`. This will:
+
+- recompile the application if that somehow hasn't already happened via your editor
+- build the `index.html` page to serve
+- copy everything in `granary-ui/public` into the magic directory under `api/src/main/resources/`.
+
+Hot reloading is disabled because we're serving the application from Http4s instead of with Webpack,
+but you don't have to restart the server when you make changes and compilation is _really_ fast, so
+it's really not so bad.
+
+#### Editor setup
+
+There are a number of good editor integrations available for Elm. The one I'm
+most used to is the [VSCode plugin](https://marketplace.visualstudio.com/items?itemName=Elmtooling.elm-ls-vscode), which
```

Comment:
This _mostly_ worked, I got some errors and when I looked in the debug logs it was because the server couldn't write to `granary-ui/elm-stuff`. When I looked it was owned by `root` - I'm not sure why that was the case. Once I changed the permissions everything worked well though.

Reply:
Huh, it probably already existed when I ran with docker for the first time. I have an idea for how to fix that

Reply:
The language server right? Not the API server? I just set up a fresh granary with the change in 7287681 and had a functioning language server the first time I opened an elm file.

Reply:
The language server was the issue yeah. I think this solution might work then so the directory can still be written to.